### DISCUSSION TITLE
Eval json problem, don't show any statistic.

### DISCUSSION THREAD

**Discussion by claudio9russo7** [obss/sahi#652] [Q&A] [answered]
I ran the "predict" command providing the .json file associated with the photos:

!sahi predict  --source /content/drive/MyDrive/ColabNotebooks/WeedNew/Test --dataset_json_path /content/drive/MyDrive/ColabNotebooks/WeedNew/Test/test.json  --model_type yolov5 --model_path /content/drive/MyDrive/ColabNotebooks/WeedNew/weed6.pt --slice_height 640 --slice_width 640 --export_visual --overlap_height_ratio 0.4 --overlap_width_ratio 0.4  --export_pickle

Then i use the coco evaluate

!sahi coco evaluate --dataset_json_path /content/drive/MyDrive/ColabNotebooks/WeedNew/Test/test.json  --result_json_path /content/runs/predict/exp2/result.json 

But i get this: 

![eval](https://user-images.githubusercontent.com/99536010/163787501-ee9098a9-5771-4f85-a1b9-2f43daf6d876.png)

Viewing the visuals with ground truth I see errors, but the results in general are excellent, I don't understand why it gives me those numbers.

**Comment by claudio9russo7** [comment]
Here some images with GT
![Optimized-IMG_20220127_112624](https://user-images.githubusercontent.com/99536010/163792178-2a8ebecb-1754-425d-a231-2918d451f044.png)
![Optimized-IMG_20220127_112641](https://user-images.githubusercontent.com/99536010/163792272-3db2ca40-3ddb-424f-89ef-6cf060f9fc38.png)

**Comment by fcakyon** [comment, marked as answer]
@claudio9russo7 have you your dataset.json category ids are properly set, see this discussion on how to update your dataset categories: https://github.com/obss/sahi/issues/273#issuecomment-992317552

**Comment by claudio9russo7** [comment]
Thanks for the quick reply, I checked as you said. Actually the ids were wrong by 1. I fixed the problem, making them match, but unfortunately the problem persists.
![eval1](https://user-images.githubusercontent.com/99536010/163890269-0101e85a-544e-404a-9b44-7c17d89ea7a6.png)

Training the model I got a mAP_0.5 of 0.3, and from the photos I see it does a good job. I don't understand these statistics
EDIT:
ok, i re-checked and didn't fix the problem respect to the dataset .json.
I saw what you linked to me but in that thread this link doesn't work anymore: "https://github.com/obss/sahi/blob/main/docs/COCO.md"

**Comment by claudio9russo7** [comment]
I solved the category_id problem, now I'm sure. As you said it was the only problem, in fact now I have consistent numbers.
Thanks @fcakyon.
![eval2](https://user-images.githubusercontent.com/99536010/163947215-9feedc6e-161b-49c5-9cc4-4cad099732ad.png)

**Comment by w615933060** [comment]
I made the same mistake as you. I checked my json file, but I didn't find any mistake. May I ask what was wrong with your json file

**Reply by claudio9russo7** [reply]
@w615933060  For me the .json file from the .txt format annotation of yolo was a category id behind, so there was a mismatch between this file and the predict file of sahi. You need to be sure about this.

**Reply by w615933060** [reply]
You mean at the end of the.json file, and how did you change it, delete it?

**Reply by claudio9russo7** [reply]
Sorry, english it's not my first language. From .txt annotation of yolo i create a .json file with some category id. When you run sahi in predict, you obtain a result.json. I just checked the category id from my json file and sahi and there was a mismatch. I change the id from my json and just works. Now, if you transform your txt yolo (if you are using yolo) with some library you need to change the code associated with the category id.

**Reply by w615933060** [reply]
I see. Thank you very much for your patience
